aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+  +------+----------------------------+----------------------------------------------------------------------------------------------------------+
+
+With the exception of R_ARM_COPY all dynamic relocations require that the place
+being relocated is a word-aligned 32-bit object.
+
+R_ARM_JUMP_SLOT is used to mark code targets that will be executed.  On
+platforms that support dynamic binding the relocations may be performed lazily
+on demand.  The unresolved address stored in the place will initially point to
+the entry sequence stub for the dynamic linker and must be adjusted during
+initial loading by the offset of the load address of the segment from its link
+address.  Addresses stored in the place of these relocations may not be used
+for pointer comparison until the relocation has been resolved.  In a REL form
+of this relocation the addend, A, is always 0.
+
+R_ARM_COPY may only appear in executable objects where e_type is set to
+ET_EXEC.  The effect is to cause the dynamic linker to locate the target symbol
+in a shared library object and then to copy the number of bytes specified by
+the st_size field to the place.  The address of the place is then used to
+pre-empt all other references to the specified symbol.  It is an error if the
+storage space allocated in the executable is insufficient to hold the full copy
+of the symbol.  If the object being copied contains dynamic relocations then
+the effect must be as if those relocations were performed before the copy was
+made.
+
+.. note::
+   R_ARM_COPY is normally only used in SVr4 type environments where the
+   executable is not position independent and references by the code and
+   read-only data sections cannot be relocated dynamically to refer to an
+   object that is defined in a shared library.
+
+   The need for copy relocations can be avoided if a compiler generates all
+   code references to such objects indirectly through a dynamically relocatable
+   location, and if all static data references are placed in relocatable
+   regions of the image.  In practice, however, this is difficult to achieve
+   without source-code annotation; a better approach is to avoid defining
+   static global data in shared libraries.
+
+Deprecated relocations
+~~~~~~~~~~~~~~~~~~~~~~
+
+Deprecated relocations are in the process of being retired from the
+specification and may be removed or marked obsolete in future revisions.  An
+object file containing these codes is still conforming, but producers should be
+changed to use the new alternatives.
+
+The relocations R_ARM_GNU_VTENTRY and R_ARM_GNU_VTINHERIT have been used by
+some toolchains to facilitate unused virtual function elimination during
+linking.  This method is not recommended and these relocations may be made
+obsolete in a future revision of this specification.  These relocations may be
+safely ignored.
+
+.. table:: Deprecated relocations
+  :widths: 35 65
```

Comment:
as described above on the version history table, please use a class to specify the width